CMakeLists.txt

```diff
@@ -370,8 +405,11 @@ if (USE_INCLUDE_WHAT_YOU_USE)
 endif ()
 
 if (ENABLE_TESTS)
-    message (STATUS "Tests are enabled")
+    message (STATUS "Unit tests are enabled")
```

Comment:
For some reason, cmake ignores this option:
in fast test: https://github.com/ClickHouse/ClickHouse/blob/master/docker/test/fasttest/run.sh#L86
in cmake log: 
```
2020-09-20 01:33:37 -- Using libcxxfs: 
2020-09-20 01:33:37 -- Using libcxxabi: cxxabi
2020-09-20 01:33:37 -- Unit tests are enabled
2020-09-20 01:33:37 -- Building for: Linux-4.9.151-35 x86_64 x86_64-linux-gnu ;
2020-09-20 01:33:37     USE_STATIC_LIBRARIES=ON
2020-09-20 01:33:37     MAKE_STATIC_LIBRARIES=ON
2020-09-20 01:33:37     SPLIT_SHARED=
2020-09-20 01:33:37     UNBUNDLED=OFF
2020-09-20 01:33:37     CCACHE=/usr/bin/ccache 3.7.3
```
Probably this check wasn't useless https://github.com/ClickHouse/ClickHouse/pull/14711/files#diff-0bb1162dc0a7bfe540790a362ee6ff96L1

Reply:
This check is not semantically correct as it's not related to ENABLE_TESTS (and user would think the latter really enables tests, whereas they are enabled by ENABLE_GTEST_LIBRARY/USE_LIBRARIES).